platform/configurator/configurator-api/src/main/java/org/codice/ddf/admin/configurator/Result.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator;
+
+import java.util.Optional;
+
+/**
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface Result {
+    boolean isTxactSucceeded();
+
+    Status getStatus();
+
+    Optional<Throwable> getBadOutcome();
```

Comment:
The problem I have is that the `Status` is _almost_ orthogonal to this failure exception. It's an enum indicating what happened; in the case of failure, this field will have a value in it. The `Outcome` you suggest is what this `Result` class _is_. `Result.getStatus()` returns the good/bad/etc. and this method is here for extracting an exception if available.

I'm wide open to suggestions on naming.